libs/estdlib/src/inet.erl

```diff
@@ -98,11 +100,22 @@ peername(Socket) ->
 %% Internal operations
 %%
 
-%% @private
-call(Socket, Msg) ->
+call(Port, Msg) ->
     Ref = erlang:make_ref(),
-    Socket ! {self(), Ref, Msg},
+    % This is `port` in feature/smp branch
+    MonitorRef = monitor(process, Port),
+    Port ! {self(), Ref, Msg},
+    Result =
+        receive
```

Comment:
Should we add this monitor to the port.erl module and support all ports this way?

Reply:
There potentially is a race condition in `port:call/2,3` indeed.
Let's fix it.